### ISSUE TITLE
Sign in to Azure CLI not working

### ISSUE THREAD

**Issue by boukepieter** [irarainey/vscode-applicationregistrations#46]
![image](https://github.com/irarainey/vscode-applicationregistrations/assets/5870796/72d278a4-9ec8-4cdd-a97d-7b88384d1c91)
I use the option to sign in, specify the tenant id and the login seems to go fine (it opens in the browser, logs in successfully and when I do az login in the terminal in vscode it gives the right respons). The application registrations shows 'initializing extension' but then the above image appears and I am not logged in. 
I am using Windows 10 Enterprise. VS Code version 1.84.1. Azure CLI version 2.53.1. Azure Application Registrations version 0.13.0.

**Comment by irarainey** [comment]
Thanks for this. I'll look into it. If you are already logged into the Az CLI in the terminal, does the extension work as expected?

**Comment by boukepieter** [comment]
No, the same thing happens.

**Comment by boukepieter** [comment]
BTW. The reason I wanted to make use of this extension is this:
![image](https://github.com/irarainey/vscode-applicationregistrations/assets/5870796/cecc14c7-c2b3-4d2b-8992-dda7c71a8e1f)

Using Azure Portal is blocked (and it advices to use VS). Could this have something to do with it?

**Comment by irarainey** [comment]
Hmmm, I'm guessing so. Ultimately behind the scenes the extension uses the same APIs that the portal does. Looking at the logic, the sign-in message is shown again if there is a problem signing in and obtaining the access token required.

The process driven by the extension is only to kick off the command `az login --tenant <name>`. Are you able to successfully login manually in a terminal?

**Comment by irarainey** [comment]
Hmmm, I'm guessing so. Behind the scenes the extension uses the same APIs that the portal does. Looking at the logic, the sign-in message is shown again if there is a problem signing in and obtaining the access token required.

The process driven by the extension is only to kick off the command `az login --tenant <name>`. Are you able to successfully login manually in a terminal?

If so, can you obtain an access token for Graph API? Ultimately that is what is happening behind the scenes. Can you run this: `az account get-access-token --scope https://graph.microsoft.com/.default`

My guess is that you are not authorised to obtain a token with that scope, and it will throw an error. The extension is likely just masking that with a failure. I guess I could amend it to better display such issues.

**Comment by boukepieter** [comment]
That set me on the right path!
az login --tenant <name> always worked fine but when I tried to get the Graph API access token I got this:
![image](https://github.com/irarainey/vscode-applicationregistrations/assets/5870796/c939c93e-9c6c-44ab-8608-35fc8bbf3ecc)
When I did az login --scope https://graph.microsoft.com/.default it turned out that firefox didn't have the right setting enabled (allowing SSO for microsoft, see https://support.mozilla.org/en-US/kb/windows-sso). When I enabled that and again did az login with the graph scope I then could get the access token and also now the sign in function of the extension is working.
